stream-chat-android-client/src/main/java/io/getstream/chat/android/client/utils/message/MessageUtils.kt

```diff
@@ -78,8 +78,28 @@ public fun Message.isDeleted(): Boolean = deletedAt != null
 /**
  * @return If the message is pinned and not deleted.
  */
+@Deprecated(
+    message = "This function is deprecated, please use isPinned() instead",
+    replaceWith = ReplaceWith("isPinned()"),
+)
 public fun Message.isPinnedAndNotDeleted(): Boolean = pinned && !isDeleted()
 
+/**
+ * @return If the message is a valid pinned message.
+ *
+ * @param now A function that provides the current time in milliseconds.
+ */
+@InternalStreamChatApi
+public inline fun Message.isPinned(
+    now: () -> Long = { System.currentTimeMillis() },
+): Boolean = pinned && !isDeleted() && !isPinExpired(now)
+
+/**
+ * @return If the message is a valid pinned message.
+ */
+@InternalStreamChatApi
+public inline fun Message.isPinExpired(now: () -> Long): Boolean = pinExpires?.let { it.time > now() } ?: false
```

Comment:
```suggestion
private inline fun Message.isPinExpired(now: () -> Long): Boolean = pinExpires?.let { it.time > now() } ?: false
```

Could we keep private this function?
At the end it is not used outside

Reply:
Can't make it `private` cause it's `inline` function, so I turned this into `internal` with `@PublishedApi` annotation.